Compile bug: error C2664: 'LSTATUS RegQueryValueExA(HKEY,LPCSTR,LPDWORD,LPDWORD,LPBYTE,LPDWORD)': cannot convert argument 2 from 'const wchar_t [20]' to 'LPCSTR'

### Git commit

$ git rev-parse HEAD
b9ab0a4d0b2ed19effec130921d05fb5c30b68c5

### Operating systems

Windows

### GGML backends

CPU

### Problem description & steps to reproduce

I'm using library [lcpp ^0.0.4](https://pub.dev/packages/lcpp).
$ git rev-parse HEAD
b9ab0a4d0b2ed19effec130921d05fb5c30b68c5

Compile error
```
ggml\src\ggml-cpu\ggml-cpu.cpp(293,21): error C2664: 'LSTATUS RegQueryValueExA(HKEY,LPCSTR,LPDWORD,LPDWORD,LPBYTE,LPDWORD)': cannot convert argument 2 from 'const wchar_t [20]' to 'LPCSTR'
```

Fix:
`RegQueryValueExA` should be changed to `RegQueryValueEx`  in lines 286 & 293.

`RegQueryValueExA` is `char*`, `RegQueryValueExW` is for `wchar*` and `RegQueryValueEx` will resolve to correct function based on setting `-DUNICODE -D_UNICODE`


### First Bad Commit

_No response_

### Compile command

```shell
flutter build windows --debug
```

### Relevant log output

```shell
\windows\flutter\ephemeral\.plugin_symlinks\lcpp\src\llama_cpp\ggml\src\ggml-cpu\ggml-cpu.cpp(286,17): error C2664: 'LSTATUS RegQueryValueExA(HKEY,LPCSTR,LPDWORD,LPDWORD,LPBYTE,LPDWORD)': cannot convert argument 2 from 'const wchar_t [20]' to 'LPCSTR' \build\windows\x64\plugins\lcpp\shared\ggml\src\ggml-cpu.vcxproj]
\windows\flutter\ephemeral\.plugin_symlinks\lcpp\src\llama_cpp\ggml\src\ggml-cpu\ggml-cpu.cpp(293,21): error C2664: 'LSTATUS RegQueryValueExA(HKEY,LPCSTR,LPDWORD,LPDWORD,LPBYTE,LPDWORD)': cannot convert argument 2 from 'const wchar_t [20]' to 'LPCSTR' \build\windows\x64\plugins\lcpp\shared\ggml\src\ggml-cpu.vcxproj]
```

--- a/ggml/src/ggml-cpu/ggml-cpu.cpp
+++ b/ggml/src/ggml-cpu/ggml-cpu.cpp
@@ -283,15 +283,15 @@ struct ggml_backend_cpu_device_context {
                         KEY_READ,
                         &hKey) == ERROR_SUCCESS) {
             DWORD cpu_brand_size = 0;
-            if (RegQueryValueEx(hKey,
-                                TEXT("ProcessorNameString"),
+            if (RegQueryValueExA(hKey,
+                                "ProcessorNameString",
                                 NULL,
                                 NULL,
                                 NULL,
                                 &cpu_brand_size) == ERROR_SUCCESS) {
                 description.resize(cpu_brand_size);
-                if (RegQueryValueEx(hKey,
-                                    TEXT("ProcessorNameString"),
+                if (RegQueryValueExA(hKey,
+                                    "ProcessorNameString",
                                     NULL,
                                     NULL,
                                     (LPBYTE)&description[0], // NOLINT

--- a/ggml/src/ggml-cpu/ggml-cpu.cpp
+++ b/ggml/src/ggml-cpu/ggml-cpu.cpp
@@ -283,14 +283,14 @@ struct ggml_backend_cpu_device_context {
                         KEY_READ,
                         &hKey) == ERROR_SUCCESS) {
             DWORD cpu_brand_size = 0;
-            if (RegQueryValueExA(hKey,
+            if (RegQueryValueEx(hKey,
                                 TEXT("ProcessorNameString"),
                                 NULL,
                                 NULL,
                                 NULL,
                                 &cpu_brand_size) == ERROR_SUCCESS) {
                 description.resize(cpu_brand_size);
-                if (RegQueryValueExA(hKey,
+                if (RegQueryValueEx(hKey,
                                     TEXT("ProcessorNameString"),
                                     NULL,
                                     NULL,

